app/ts/library/ethereum.ts

```diff
@@ -0,0 +1,22 @@
+import { BigNumber, ethers } from 'ethers'
+import { TransactionReceipt, TransactionResponse, TransferTransactionResponse } from '../types.js'
+import { ERC20ABI } from './ERC20ABI.js'
+
+export function isTransferTransaction(txResponse: TransactionResponse): txResponse is TransferTransactionResponse {
+	return txResponse.data.startsWith('0xa9059cbb')
+}
+
+export function getTransferTokenValue(transactionReceipt: TransactionReceipt): BigNumber {
+	const IERC20 = new ethers.utils.Interface(ERC20ABI)
+
+	let tokenValue = undefined
+
+	for (const receiptLog of transactionReceipt.logs) {
+		const parsedLog = IERC20.parseLog(receiptLog)
+		if (parsedLog.name.toLowerCase() === 'transfer') {
+			tokenValue = parsedLog.args.value
+		}
```

Comment:
This appears to use the value of the last token transfer in a transaction.  Why not the first?

Even better would be for this function to take the expected sender, recipient, and token and make sure the transfer log matches everything and only extract the value when the rest match.

Alternatively, this function could return *all* of the parsed transfer logs and leave it up to the caller to decide which one to use in the case of multiple.

Reply:
i used a different approach after https://github.com/DarkFlorist/lunaria/pull/89/commits/540795b94176d74702c283d631cf134bc1aa761a

> This appears to use the value of the last token transfer in a transaction. Why not the first?

I'm not aware of this scenario so I replaced the loop with `find` so that only the first matched log will be used

Added type guards as well to limit response to only the expected transfer log result